### PR TITLE
add logging to registry flows

### PR DESCRIPTION
**Pull request by tomerd**
motivation: easier debugging

changes:
* add info logging for actions that may be interesting to users
* add debug logging to assist troubleshooting

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
cc @yim-lee

File: Sources/Basics/AuthorizationProvider.swift [hunk 1]
```diff
@@ -225,7 +225,7 @@ public class KeychainAuthorizationProvider: AuthorizationProvider, Authorization
         } catch {
             switch error {
             case AuthorizationProviderError.notFound:
-                self.observabilityScope.emit(info: "No credentials found for server \(server) in keychain")
```
Comment (yim-lee) [inline Sources/Basics/AuthorizationProvider.swift:228]: Why delete this?

Comment (tomerd) [reply Sources/Basics/AuthorizationProvider.swift:228]: it is very noisy in the log, and I did not think it provided huge amount of value. we can also tune down to debug, or write code that caches negative results so it's called less often. opinions?

Comment (yim-lee) [reply Sources/Basics/AuthorizationProvider.swift:228]: I see. Maybe debug? If that's still too noisy then let's remove it.

**Comment by tomerd** [comment]
@swift-ci smoke test

**Review by yim-lee — APPROVED** [review]

**Pull request merged**